packages/opentelemetry-test-utils/src/instrumentations/otel-provider-api.ts

```diff
@@ -18,27 +18,76 @@ import {
   ReadableSpan,
 } from '@opentelemetry/sdk-trace-base';
 
-const OTEL_TESTING_MEMORY_EXPORTER = Symbol.for(
-  'opentelemetry.testing.memory_exporter'
+import {
+  InMemoryMetricExporter,
+  ResourceMetrics,
+} from '@opentelemetry/sdk-metrics';
+
+const OTEL_TRACING_TESTING_MEMORY_EXPORTER = Symbol.for(
+  'opentelemetry.tracing.testing.memory_exporter'
+);
+
+const OTEL_METRICS_TESTING_MEMORY_EXPORTER = Symbol.for(
+  'opentelemetry.metrics.testing.memory_exporter'
 );
 
-type OTelProviderApiGlobal = {
-  [OTEL_TESTING_MEMORY_EXPORTER]?: InMemorySpanExporter;
+type OTelProvidersApiGlobal = {
+  [OTEL_TRACING_TESTING_MEMORY_EXPORTER]?: InMemorySpanExporter;
+  [OTEL_METRICS_TESTING_MEMORY_EXPORTER]?: InMemoryMetricExporter;
+};
+
+const _global = global as OTelProvidersApiGlobal;
+
+export const getTracingTestMemoryExporter = ():
+  | InMemorySpanExporter
+  | undefined => {
+  return _global[OTEL_TRACING_TESTING_MEMORY_EXPORTER];
 };
-const _global = global as OTelProviderApiGlobal;
 
-export const getTestMemoryExporter = (): InMemorySpanExporter | undefined => {
-  return _global[OTEL_TESTING_MEMORY_EXPORTER];
+export const getMetricsTestMemoryExporter = ():
+  | InMemoryMetricExporter
+  | undefined => {
+  return _global[OTEL_METRICS_TESTING_MEMORY_EXPORTER];
 };
 
-export const setTestMemoryExporter = (memoryExporter: InMemorySpanExporter) => {
-  _global[OTEL_TESTING_MEMORY_EXPORTER] = memoryExporter;
+export const setMetricsTestMemoryExporter = (
+  memoryExporter: InMemoryMetricExporter
+) => {
+  _global[OTEL_METRICS_TESTING_MEMORY_EXPORTER] = memoryExporter;
+};
+
+export const setTracingTestMemoryExporter = (
+  memoryExporter: InMemorySpanExporter
+) => {
+  _global[OTEL_TRACING_TESTING_MEMORY_EXPORTER] = memoryExporter;
 };
 
 export const getTestSpans = (): ReadableSpan[] => {
-  return getTestMemoryExporter()!.getFinishedSpans();
+  return getTracingTestMemoryExporter()!.getFinishedSpans();
+};
+
+export async function getTestMetrics(
+  numberOfExports: number
+): Promise<ResourceMetrics[]> {
+  if (numberOfExports <= 0) {
+    throw new Error('numberOfExports must be greater than or equal to 0');
+  }
+
+  const exporter = getMetricsTestMemoryExporter()!;
+  let totalExports = 0;
+  while (totalExports < numberOfExports) {
+    await new Promise(resolve => setTimeout(resolve, 20));
```

Comment:
See https://github.com/open-telemetry/opentelemetry-js/pull/3242, we should be able to implement a deterministic MetricReader that performs metric collection every time `getTestMetrics` is called.

Reply:
Wow awesome! Thanks!
Added TODO to use this reader.

Reply:
The reader in the PR I referenced is not going to be exported since it is used in ad-hoc tests. I believe we can create a subclass of MetricReader in this test-utils package instead, to provide basic testing features.

Reply:
Oh :/ Should we ask them to move it to the src? similar to the InMemoryExporters.
I can copy paste the code but I think it's better to implement it in one place

Reply:
No, it is not part of the spec so I believe we are not going to expose it in the public interfaces. I think duplicating the work here is fine since the interfaces exposed here are going to be public, whilst the test facilities in that PR are not going to be exposed as public interfaces.

Reply:
Oh, that makes sense. I'll implement it here. Thanks for quick response :)